models/models.js

```diff
@@ -1,12 +1,112 @@
-var mongoose = require("mongoose");
-var userSchema = new mongoose.Schema({
-    username: String,
-    email: String,
-    password: String,
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+const bcrypt = require('bcrypt-nodejs');
+
+
+
+let usernameLengthChecker = (username) => {
+    if (!username) {
+        return false;
+    }
+    if (username.length < 3 || username.length > 15) {
+        return false;
+    }
+    return true;
+};
+
+let validUsername = (username) => {
+    if (!username) {
+        return false;
+    }
+    const reg = new RegExp(/^[a-zA-Z0-9]+$/);
+    return reg.test(username);
+};
+
+const usernameValidators = [{
+        validator: usernameLengthChecker,
+        message: 'Username must be at least 3 characters but no more than 15'
+    },
+    {
+        validator: validUsername,
+        message: 'Username must not have any special characters'
+    }
+];
+
+
+
+let emailLengthChecker = (email) => {
+    if (!email) {
+        return false;
+    }
+    if (email.length < 5 || email.length > 30) {
```

Comment:
Incorrect max length 30: https://stackoverflow.com/questions/386294/what-is-the-maximum-length-of-a-valid-email-address